internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -1347,18 +1404,26 @@ func expandAgentPoolUpgradeSettings(input []interface{}) *agentpools.AgentPoolUp
 	if maxSurgeRaw := v["max_surge"].(string); maxSurgeRaw != "" {
 		setting.MaxSurge = pointer.To(maxSurgeRaw)
 	}
+	if maxUnavailableRaw, ok := v["max_unavailable"].(string); ok && maxUnavailableRaw != "" && maxUnavailableRaw != "0" {
+		setting.MaxUnavailable = pointer.To(maxUnavailableRaw)
+		// maxSurge must be set to 0 for maxUnavailable to be set. The two values can't both be active at the same time.
+		setting.MaxSurge = pointer.To("0")
```

Comment:
If `max_unavailable` is specified, per the schema definition, `max_surge` won't be set. Then shouldn't `setting.MaxSurge` be omit in the request? Or does API requires `setting.MaxSurge` be explicitly set to 0?

Reply:
The API requires `setting.MaxSurge` be explicitly set to 0